src/languages/smtlib/parseSmtlib.mly

```diff
@@ -137,60 +170,109 @@ command_option:
 ;
 
 command:
-  | OPEN POP n=NUMERAL CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.pop ~loc (int_of_string n) }
-  | OPEN PUSH n=NUMERAL CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.push ~loc (int_of_string n) }
-
   | OPEN ASSERT t=term CLOSE
     { let loc = L.mk_pos $startpos $endpos in S.assert_ ~loc t }
   | OPEN CHECK_SAT CLOSE
     { let loc = L.mk_pos $startpos $endpos in S.check_sat ~loc () }
 
-  | OPEN SET_LOGIC s=SYMBOL CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.set_logic ~loc s }
+  | OPEN DECLARE_CONST s=SYMBOL ty=sort CLOSE
+    { let id = I.(mk term s) in
+      let loc = L.mk_pos $startpos $endpos in
+      S.fun_decl ~loc id [] ty }
 
-  | OPEN GET_INFO i=KEYWORD CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.get_info ~loc i }
-  | OPEN SET_INFO c=command_option CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.set_info ~loc c }
+  | OPEN DECLARE_DATATYPE s=SYMBOL OPEN clist=datatype_dec CLOSE CLOSE
+    { let id = I.(mk sort s) in
+      let loc = L.mk_pos $startpos $endpos in
+      S.dtype_decl ~loc id [] (snd clist) }
```

Comment:
Maybe assert here that `(fst clist) = []` ? (probably splitting `clist` with a let first), to make it clear that no information is discarded.